### PR TITLE
Refactor/scripts.js

### LINKED ISSUE #9: `script.js`
- [x] Line 3: ingredientData variable isn't being used anywhere
- [x] Line 11-24: all the document.querySelector() variables should be const. Also, could use getElementById for some.
- [x] Could re-organize document selectors into objects for better readability
- [x] Could cut down on event listeners by using a helper function for all `'load'`, `'click'` events
- [x] There is a mix of double & single quotes being used
- [x] The full images folder isn't being loaded when the page loads (this could be something with webpack import/export stuff)
- [x] The src attributes for img tags is using too many dots (should only have to be `./images/<file name>`
- [x] Horrible organization on lines 11-25
- [x] Line 152: event is not passed in as param in function definition, also looks like it could be better off as switch/case instead of so many if/elses

### PR DESCRIPTION
**Pull request by mistercanderson**
# Description of Changes

- Changed webpack entry point to `index.js`
- Refactored multiple functions for better modularity/readability
- 'click' functions are now stored in helper function to reduce total eventListeners
- Re-organized unnecessary global variables
- Added ability for "show all recipes" banner to appear whenever filtering/search functions are called
- Quote usage is now consistent to single quotes for JS & double for HTML attributes
- Plugged in hardcoded data for the sake of being able to work with JS functions
- Refactored some of the `apiCalls` fetch requests

# Linked Issues

- closes #9

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -31,34 +30,35 @@ import './images/seasoning.png'
...
 const pantryInfo = [];
 const recipes = [];
-const savedRecipesBtn = document.querySelector('.saved-recipes-btn');
-const searchBtn = document.querySelector('.search-btn');
 const searchForm = document.querySelector('#search');
 const searchInput = document.querySelector('#search-input');
-const showPantryRecipes = document.querySelector('.show-pantry-recipes-btn');
 const tagList = document.querySelector('.tag-list');
-let user;
+const buttons = {
```
Comment (rhen92) [inline src/scripts.js:40]: Love the way you stored these document.querySelector(s) in a variable!

File: src/scripts.js [hunk 2]
```diff
@@ -67,9 +67,32 @@ function onLoad() {
   generateUser();
 }
 
+function buttonHandlers(event) {
```
Comment (rhen92) [inline src/scripts.js]: Good use of switch!!!

**Comment by rhen92** [comment]
Yes lets figure out how to get these fetch requests to play nicely with us! Great work!

**Review by rhen92 — APPROVED** [review]

File: src/index.js [hunk 3]
```diff
@@ -3,8 +3,11 @@
 
 // An example of how you tell webpack to use a CSS (SCSS) file
 import './css/base.scss';
+import './recipe.js';
```
Comment (battan40) [inline src/index.js:6]: We'll review here to get our api calls going, in the next few days, good call to keep the data coming in, for now, from elsewhere.

File: src/scripts.js [hunk 4]
```diff
@@ -27,50 +31,71 @@ import './images/seasoning.png'
 import './css/base.scss';
 import './css/styles.scss';
 
-import User from './user';
-import Recipe from './recipe';
-
-let allRecipesBtn = document.querySelector(".show-all-btn");
```
Comment (battan40) [inline src/scripts.js:33]: nice work down sizing the file

File: webpack.config.js [hunk 5]
```diff
@@ -2,7 +2,7 @@ const path = require('path');
 const HtmlWebpackPlugin = require('html-webpack-plugin');
 
 module.exports = {
-  entry: './src/scripts.js',
+  entry: './src/index.js',
```
Comment (battan40) [inline webpack.config.js:5]: we'll revisit this later, as well, to verify this is the way we want to go

**Review by battan40 — APPROVED** [review]

**Pull request merged**